pkg/analysis/passes/sponsorshiplink/sponsorshiplink.go

```diff
@@ -0,0 +1,53 @@
+package sponsorshiplink
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+)
+
+var (
+	sponsorshiplink = &analysis.Rule{Name: "sponsorshiplink", Severity: analysis.Recommendation}
+	explanation     = "Consider to add a sponsorship link in your plugin.json file (Info.Links section), which will be shown on the plugin details page to allow users to support your work if they wish."
```

Comment:
I'd also add they should use "sponsor" or "sponsorship" as the link name otherwise it won't be picked up